ons-spark/spark-analysis/interpolation.md

```diff
@@ -0,0 +1,502 @@
+2021-01-26	A	        120.0	 0	       120.000000
+2021-01-31	A	        NaN	   1	       125.000000
+2021-02-05	A	        130.0	 0	       130.000000
+```
+```{code-tab} plaintext R output
+   area_code period     count impute_flag count_final
+   <chr>     <date>     <dbl>       <dbl>       <dbl>
+ 1 A         2021-01-01   100           0       100  
+ 2 A         2021-01-06    NA           1       110  
+ 3 A         2021-01-11    NA           1       120  
+ 4 A         2021-01-16   130           0       130  
+ 5 A         2021-01-21   100           0       100  
+ 6 A         2021-01-26   120           0       120  
+ 7 A         2021-01-31    NA           1       125  
+ 8 A         2021-02-05   130           0       130  
+ 9 B         2021-01-01    NA           1        NA  
+10 B         2021-01-11    85           0        85  
+11 B         2021-01-19    82           0        82  
+12 B         2021-01-31    75           0        75  
+13 B         2021-02-10    NA           1        81.7
+14 B         2021-02-15    85           0        85  
+15 B         2021-02-20    NA           1        80  
+16 B         2021-02-25    75           0        75  
+```
+````
+
+
+Note that the first value for `area_code` B could not be interpolated because there is no preceding count available. In other words, we would need to extrapolate to estimate for this value.
+
+Once again, we will plot the series with pandas/ggplot. The original data are plotted as red circles and the final series in blue.
+
+````{tabs}
+```{code-tab} py
+pdf[pdf["area_code"] == "A"].drop("impute_flag", axis=1).plot(style={"count":"ro", "count_final":"b+-"}, alpha=0.5)
+pdf[pdf["area_code"] == "B"].drop("impute_flag", axis=1).plot(style={"count":"ro", "count_final":"b+-"}, alpha=0.5)
+```
+````
+
+```{figure} ../images/interpolation_pyplot_afterA.png
+---
+width: 80%
+name: Plot of pdf in Python
+alt: Chart showing value of count for area "A" above and area "B" below. All values are now connected by a line. 
+---
+```
+```{figure} ../images/interpolation_pyplot_afterB.png
+---
+width: 80%
+name: Plot of pdf in Python
+alt: Chart showing value of count for area "A" above and area "B" below. All values are now connected by a line. 
+---
+Interpolated plot of pdf in Python split by area code "A" (top) and area code "B" (bottom).
+```
+
+````{tabs}
+```{code-tab} r R
+ggplot(df, mapping = aes(x = period, y = count_final)) +
+  geom_point(aes(colour = factor(impute_flag))) +
+  scale_fill_manual(values = c("red", "blue")) +
+  geom_line() +
+  labs(colour = "impute_flag") +
+  facet_wrap(~area_code, nrow = 2)
+```
+````
+
+```{figure} ../images/interpolation_rplot_after.png
+---
+width: 80%
+name: Plot of df in R
+alt: Chart showing value of count for area "A" above and area "B" below. All values are now connected by a line.
+---
+Interpolated plot of df in R split by area code "A" (top) and area code "B" (bottom).
+```
+
+
+## Further resources
+
```

Comment:
Thanks! I've added a link to the window functions page and to the visualisation page now. I'm not sure if there are any others that would be good to include? There might be some more to add eventually too as the Analysis in Spark section continues to grow.

Reply:
After Checking with Nathan,  the URLs for book pages needs to be defined using the relative path '../spark-functions/window-functions' etc. This is just in case the book URL changes at any point in the future. Everything else looks good to merge once update :)